src/main/target/SITL/target.c

```diff
@@ -221,34 +270,43 @@ void failureMode(failureMode_e mode) {
 
 // Time part
 // Thanks ArduPilot
+uint64_t nanos64_real() {
+	struct timespec ts;
+	clock_gettime(CLOCK_MONOTONIC, &ts);
+	return (ts.tv_sec*1e9 + ts.tv_nsec) - (start_time.tv_sec*1e9 + start_time.tv_nsec);
+}
 uint64_t micros64_real() {
 	struct timespec ts;
 	clock_gettime(CLOCK_MONOTONIC, &ts);
-	return 1.0e6*((ts.tv_sec + (ts.tv_nsec*1.0e-9)) -
-			(start_time.tv_sec +
-			(start_time.tv_nsec*1.0e-9)));
+	return 1.0e6*((ts.tv_sec + (ts.tv_nsec*1.0e-9)) - (start_time.tv_sec + (start_time.tv_nsec*1.0e-9)));
 }
 uint64_t millis64_real() {
 	struct timespec ts;
 	clock_gettime(CLOCK_MONOTONIC, &ts);
-	return 1.0e3*((ts.tv_sec + (ts.tv_nsec*1.0e-9)) -
-			(start_time.tv_sec +
-			(start_time.tv_nsec*1.0e-9)));
+	return 1.0e3*((ts.tv_sec + (ts.tv_nsec*1.0e-9)) - (start_time.tv_sec + (start_time.tv_nsec*1.0e-9)));
 }
 
 uint64_t micros64() {
-	static uint64_t last_time = 0;
-	uint64_t now = micros64_real();
-	uint64_t out = last_time + ((now - last_time) * simRate);
-	last_time = out;
-	return  out;
+	static uint64_t last = 0;
+	static uint64_t out = 0;
+	uint64_t now = nanos64_real();
+
+	out += (now - last) * simRate;
+	last = now;
+
+	return out*1e-3;
+//	return micros64_real();
 }
 uint64_t millis64() {
```

Comment:
This should be definitely derived from micros64. 
As it is now, static out in both functions may get out of sync, which may cause 'interesting' consequences ...